### ISSUE TITLE
Scope of variables

### ISSUE THREAD

**Issue by kubik369** [cypress-io/cypress#182]
**Description**
When running all the tests, if two test files have the same variable defined at the top, the variable is rewritten by the contents of the variable in the test which goes second, resulting in the failure of the first test. If the tests are ran separately, they both pass. Workaround is to use different variable names or define the variables inside of the first describe(), but it should be mentioned somewhere that running all the tests basically concatenates all the test files into one big file with a single global scope (my excuses for this issue if it is known, but I could not find it anywhere :/ ).

**Code**
_01-test_

```
var foo = '#foo'
describe('test one', function() {
    it('should work', function() {
        cy.get(foo)
    })
})
```

_02-test_

```
var foo = '#bar'
describe('test two', function() {
    it('should work', function() {
        cy.get(foo)
    })
})
```

**Steps To Reproduce**
- webpage for the first test should contain element with #foo and second page should contain element with #bar.
- run the tests separately, everything passes
- run all the tests and the first test fails, because it tries to find an element from the second page

**Comment by jennifer-mann** [comment]
When you run "all" the tests, Cypress is not concatenating anything, it's simply serving more `<script>` tags into window. When you're setting `var foo`, you're in the global scope (because you're not in a function). Basically this would behave identically as any other webpage where you have global variable collisions. You're creating `window.foo` in the first script tag, and then in the 2nd script tag you're replacing `window.foo` with something else.

If you wanted variables to be private you would simply wrap your tests in an IFFE like `(function(){})()`.
However the deeper issue here that you're bringing up is the confusion around how Cypress builds your spec files and what the best practice should be around sharing logic between spec files.

We intend to release a version in the future that supports ES6 / CommonJS which will automatically wrap all spec files and prevent global leakage.

**Comment by brian-mann** [comment]
This was effectively fixed in `0.18.0` when we added babel/ES6 support.
